shoes-swt/lib/shoes/swt/common/painter.rb

```diff
@@ -66,6 +66,15 @@ def draw_setup(graphics_context)
         def draw(_graphics_context)
         end
 
+        def needs_rotate?
+          dsl = @obj.dsl
+          supports_rotate?(dsl) && dsl.rotate != 0
+        end
+
+        def supports_rotate?(dsl)
+          dsl.is_a?(::Shoes::Oval) || dsl.is_a?(::Shoes::Rect)
+        end
```

Comment:
This should eventually move up to be an instance method or something else so we don't do class checks :)